src/components/TextField/TextField.tsx

```diff
@@ -20,12 +22,17 @@ export const TextField: React.FC<Props> = ({
   required = false,
   onChange = () => {},
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
-  const [touched, setToched] = useState(false);
+  const [touched, setTouched] = useState(false);
+  const correctValue = getValidUrl(pattern, value);
   const hasError = touched && required && !value;
+  const hasValidateError
+    = touched
+    && value
+    && !correctValue
+    && (name === 'imdbUrl'
+      || name === 'imgUrl');
```

Comment:
```suggestion
  const hasError = touched && required && !value;
  const hasValidateError = (
    name === 'imdbUrl'
      || name === 'imgUrl'
    )
    && touched
    && value
    && !checkIsValidUrl(pattern, value);
```

Don't perform unnecessary actions